### PR TITLE
Time and Number-range detection test cases in yaml  format

### PR DESCRIPTION
**Pull request by ashutoshsingh0223**
Change the format of file containing test cases from CSV to YAML for time and num-range detection

### CODE REVIEW

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
+tests:
+  en:
+#  - _id_: en_1
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: Why are some of these commented?

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 2]
```diff
@@ -1,254 +1,84 @@
...
+            number_detector_object = NumberDetector(entity_name="number", language=language, unit_type=unit_type)
+            number_dicts, spans = number_detector_object.detect_entity(message)
+
+            expected_number_dicts, expected_spans = parse_expected_outputs(testcase["outputs"])
+            expected_outputs = list(six.moves.zip(expected_number_dicts, expected_spans))
+
+            prefix = failure_string_prefix.format(message=message, language=language)
+
+            self.assertEqual(len(number_dicts), len(spans),
+                             prefix + u"Returned dates and original_texts have different lengths")
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: log message is wrong. It says "dates"

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 3]
```diff
@@ -1,254 +1,84 @@
...
+            self.assertEqual(len(spans), len(expected_outputs),
+                             prefix + u"Returned dates and expected_outputs have different lengths")
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: log message is wrong. It says "dates". Same for other detectors as well

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml (hunk 1 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: They were written for a function other than `NumberDetector`.
I will do all other functions later.
This is just for detectors.

File: ner_v2/tests/numeral/number/en/test_number_detection.py (hunk 2 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/test_number_detection.py]: Done

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 4]
```diff
@@ -1,254 +1,84 @@
...
+                        cls.get_yaml_test(testcase=testcase,
+                                          language=language,)
+                    )
+
+    @classmethod
+    def get_yaml_test(cls, testcase, language, **kwargs):
+        def parse_expected_outputs(expected_outputs):
+            num_dicts, original_texts = [], []
+            for expected_output in expected_outputs:
+                if "detected_number" in expected_output:
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: Why is this needed? I am not sure where this applies

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 5]
```diff
@@ -1,254 +1,84 @@
...
+import yaml
 from django.test import TestCase
 
-
-from ner_v2.detectors.numeral.utils import get_number_from_number_word
 from ner_v2.detectors.numeral.number.number_detection import NumberDetector
-from ner_v2.detectors.numeral.number.standard_number_detector import NumberVariant
-
-
-class NumberDetectionTest(TestCase):
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: Instead of deleting this, rename this to `TestNumberFromWords` and only keep tests that test `get_number_from_number_word`

File: ner_v2/tests/numeral/number/en/test_number_detection.py (hunk 4 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/test_number_detection.py]: This check is for those commented test cases.

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 6]
```diff
@@ -1,254 +1,84 @@
...
+            for output in six.moves.zip(number_dicts, spans):
+
+                self.assertIn(output, expected_outputs,
+                              prefix + u"{got} not in {expected_outputs}".format(got=output,
+                                                                                 expected_outputs=expected_outputs))
+
+        return run_test
+
+
+class NumberDetectorTests(six.with_metaclass(NumberDetectorTestsMeta, TestCase)):
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: Use naming convention `TestWhateverDetector`, in this case `TestNumberDetector`. Same for other test classes

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+#      - output_id: "3"
+#        detected_number: 3
+#        original_text: three
+#  - _id_: en_7
+#    message: there are one thousand   one   hundred two students attending placement drive
+#    outputs:
+#      - output_id: "1"
+#        detected_number: 1102
+#        original_text: one thousand   one   hundred two
+  - _id_: en_8
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: Any reason behind choosing `_id_` instead of `id` ?

File: ner_v2/tests/numeral/number/en/test_number_detection.py (hunk 5 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/test_number_detection.py]: okay

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml (hunk 7 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: Not really. Python has `id` as a keyword

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+    message: 100 got selected for interview
+    unit_type: null
+    outputs:
+      - output_id: "1"
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: 1. Any reason why not use `id`?
2. Also any reason why these are not just ints ? like 1 ?

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        value: "100"
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: values are str ?
Are we sure this is correct ?

File: ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml [hunk 10]
```diff
@@ -0,0 +1,661 @@
...
+      min_value: '200'
+      original_text: 200 to 300
+      output_id: '1'
+      unit: null
+    unit_type: null
+  - _id_: en_11
+    message: 200 – 300
+    outputs:
+    - max_value: '300'
+      min_value: '200'
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml]: min_value and max_value are str, is this correct ?

File: ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml [hunk 11]
```diff
@@ -0,0 +1,661 @@
+tests:
+  en:
+  - _id_: en_1
+    message: I want more than 200 banana
+    outputs:
```
Comment (chiragjn) [inline ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml]: Outputs don't have ids like they do in number detector tests yaml ?

File: ner_v2/tests/pattern/phone_number/phone_number_ner_tests.yaml [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+  - _id_: en_1
+    message: Set a reminder on 02226129854
+    outputs:
+    - original_text: '02226129854'
+      output_id: '1'
+      value: '02226129854'
+  - _id_: en_2
+    message: Set a reminder on 022 26129854
+    outputs:
+    - original_text: 022 26129854
```
Comment (chiragjn) [inline ner_v2/tests/pattern/phone_number/phone_number_ner_tests.yaml]: Avoid inconsistency, some original_text are quoted, some are not.

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml (hunk 7 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: They can be.  But many things across detectors are already dealing with confusion like NumberDetector gives `strings` as `value`.

Putting language was looking more readable plus i thought it  would be easier to match outputs after these are translated.

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml (hunk 9 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: Yes. `NumberDetector` gives `value` as `str`

File: ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml (hunk 11 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml]: They do.

File: ner_v2/tests/pattern/phone_number/phone_number_ner_tests.yaml (hunk 12 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/pattern/phone_number/phone_number_ner_tests.yaml]: Okay

File: ner_v2/tests/temporal/time/time_ner_tests.yaml [hunk 13]
```diff
@@ -0,0 +1,552 @@
+args:
+    timezone: "UTC"
```
Comment (chiragjn) [inline ner_v2/tests/temporal/time/time_ner_tests.yaml]: I am not sure on this but we should probably use `Asia/Kolkata` if our test writers are going to be mostly in India

File: ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml (hunk 10 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml]: Yes.

File: ner_v2/tests/temporal/time/time_ner_tests.yaml (hunk 13 above)
Comment (ashutoshsingh0223) [reply ner_v2/tests/temporal/time/time_ner_tests.yaml]: Okay

File: ner_v2/tests/numeral/number/en/number_ner_tests.yaml (hunk 7 above)
Comment (chiragjn) [reply ner_v2/tests/numeral/number/en/number_ner_tests.yaml]: I know Python has "id" as keyword, but that shouldn't be a problem because here it would be a dict key. It is just that it would be easier to write `id` than `_id_`

File: ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml (hunk 11 above)
Comment (chiragjn) [reply ner_v2/tests/numeral/number_range/number_range_ner_tests.yaml]: Oops missed in sorting. That being said, should we sort keys? In my opinion, sorting makes it harder to read and understand the output.

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 14]
```diff
@@ -1,88 +1,224 @@
...
+            u'zero': NumberVariant(scale=1, increment=0)
+        }
+
+    def test_get_number_with_only_unit_in_number_word(self):
+        """
+        Number detection from word with only unit ex - 'one', 'two', 'twenty'
+        """
+        message = 'I want to book for one passenger'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 15]
```diff
@@ -1,88 +1,224 @@
...
+        self.assertEqual(len(zipped), 1)
+        self.assertIn((1, 'one'), zipped)
+
+    def test_get_number_with_only_scale_in_number_word(self):
+        """
+        Number detection from word with only scale like - 'hundred', 'thousand'
+        """
+        message = 'need hundred change'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 16]
```diff
@@ -1,88 +1,224 @@
...
+        self.assertEqual(len(zipped), 1)
+        self.assertIn((100, 'hundred'), zipped)
+
+    def test_get_number_with_scale_and_unit_in_number_word(self):
+        """
+        Number detection from word with scale and unit like - 'one hundred', 'one thousand two hundred five'
+        """
+        message = 'haptik get one thousand two hundred five messages daily'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 17]
```diff
@@ -1,88 +1,224 @@
...
+        self.assertEqual(len(zipped), 1)
+        self.assertIn((1205, 'one thousand two hundred five'), zipped)
+
+    def test_get_number_with_multiple_unit_number_word(self):
+        """
+        Number detection from word with multiple unit numbers like 'one two three twenty one'
+        """
+        message = 'one two three'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 18]
```diff
@@ -1,88 +1,224 @@
...
+        self.assertIn((2, 'two'), zipped)
+        self.assertIn((3, 'three'), zipped)
+
+    def test_get_number_with_multiple_unit_scale_number_word(self):
+        """
+        Number detection from word with multiple unit scale numbers like 'one thousand to one hundred two'
+        """
+        message = 'one thousand to one hundred two'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 19]
```diff
@@ -1,88 +1,224 @@
...
+        self.assertIn((1000, 'one thousand'), zipped)
+        self.assertIn((102, 'one hundred two'), zipped)
+
+    def test_get_number_with_combination_of_unit_scale_and_unit_number_word(self):
+        """
+        Number detection from word with combination of unit scale and unit number like 'one thousand one two three'
+        """
+        message = 'one thousand one two three'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/temporal/time/test_time_detection.py [hunk 20]
```diff
@@ -1,88 +1,224 @@
...
-                                                                             range_enabled=range_enabled)
-
-                self.assertEqual(len(detected_times), len(expected_times))
-                self.assertEqual(len(original_texts), len(expected_original_texts))
-
-                for detected_output_pair in zip(detected_times, original_texts):
-                    self.assertIn(detected_output_pair, expected_output)
+        message = 'there are one thousand   one   hundred two students attending placement drive'
+        detect_texts, original_texts = get_number_from_number_word(message, self.number_words_map)
+        zipped = zip(detect_texts, original_texts)
```
Comment (srajdev) [inline ner_v2/tests/temporal/time/test_time_detection.py]: W1637 [34mzip built-in referenced when not iterating[0m

File: ner_v2/tests/numeral/number/en/test_number_detection.py [hunk 21]
```diff
@@ -136,119 +138,80 @@ def test_get_number_with_multiple_spaces_in_unit_scale_number_word(self):
...
-        Number detection for english language for decimal number like '100.2'
-        """
-        message = u'Todays temperature is 11.2 degree celsius'
-        number_detector_object = NumberDetector(entity_name=self.entity_name, language='en')
-        number_dicts, original_texts = number_detector_object.detect_entity(message)
 
-        zipped = zip(number_dicts, original_texts)
-        self.assertEqual(len(zipped), 1)
-        self.assertIn(({'value': '11.2', 'unit': None}, u'11.2'), zipped)
+class TestNumberDetectorMeta(type):
```
Comment (srajdev) [inline ner_v2/tests/numeral/number/en/test_number_detection.py]: E303 too many blank lines (4)

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by haptik-deployment** [comment]
UNIT TESTS FAILED... Please resolve issues

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge

**Comment by srajdev** [comment]
:+1: No lint errors found.

**Comment by haptik-deployment** [comment]
UNIT TESTS HAVE PASSED... Good To Merge
